src/Avalonia.Controls/ProgressRing.cs

```diff
@@ -0,0 +1,102 @@
+using System;
+using Avalonia.Controls.Metadata;
+using Avalonia.Controls.Primitives;
+using Avalonia.Layout;
+using Avalonia.Media;
+
+namespace Avalonia.Controls
+{
+    /// <summary>
+    /// A control used to indicate the progress of an operation.
+    /// </summary>
+    [PseudoClasses(":preserveaspect", ":indeterminate")]
+    public class ProgressRing : RangeBase
+    {
+        public static readonly StyledProperty<bool> IsIndeterminateProperty =
+            ProgressBar.IsIndeterminateProperty.AddOwner<ProgressRing>();
+        
+        public static readonly StyledProperty<bool> PreserveAspectProperty =
+            AvaloniaProperty.Register<ProgressRing, bool>(nameof(PreserveAspect), true);
+        
+        public static readonly StyledProperty<double> ValueAngleProperty = 
+            AvaloniaProperty.Register<ProgressRing, double>(nameof(ValueAngle), -90.0);
+
+        static ProgressRing()
+        {
+            MinimumProperty.Changed.AddClassHandler<ProgressRing>(OnRangePropertiesChanged);
+            ValueProperty.Changed.AddClassHandler<ProgressRing>(OnRangePropertiesChanged);
+            MaximumProperty.Changed.AddClassHandler<ProgressRing>(OnRangePropertiesChanged);
+        }
+
+        public ProgressRing()
+        {
+            UpdatePseudoClasses(IsIndeterminate, PreserveAspect);
+        }
+
+        public bool IsIndeterminate
+        {
+            get => GetValue(IsIndeterminateProperty);
+            set => SetValue(IsIndeterminateProperty, value);
+        }
+
+        public bool PreserveAspect
+        {
+            get => GetValue(PreserveAspectProperty);
+            set => SetValue(PreserveAspectProperty, value);
+        }
+
+        public double ValueAngle
```

Comment:
If this property was added only to simplify templating, it makes sense to move it to inner property with relevant name.
See https://github.com/AvaloniaUI/Avalonia/blob/master/src/Avalonia.Controls/ProgressBar.cs#L151
So it's it will be like "TemplateProperties.ValueAngle"

Reply:
Meaning that developers shouldn't try to use it elsewhere